en/getting-started/first-game/index.md

```diff
@@ -370,173 +832,203 @@ export class GameManager extends Component {
+}
+
+init() {
+    // Active start menu
+    if (this.startMenu) {
+        this.startMenu.active = true;
+    }
+    // Generate the runway
+    this.generateRoad();
+    if(this.playerCtrl){
+        // Disable user input
+        this.playerCtrl.setInputActive(false);
+        // Reset the player's position
+        this.playerCtrl.node.setPosition(Vec3.ZERO);
     }
+}
 
-    set curState (value: GameState) {
-        switch(value) {
-            case GameState.GS_INIT:
-                this.init();
-                break;
-            case GameState.GS_PLAYING:
-                if (this.startMenu) {
-                    this.startMenu.active = false;
+set curState (value: GameState) {
+    switch(value) {
+        case GameState.GS_INIT:
+            this.init();
+            break;
+        case GameState.GS_PLAYING:
+            if (this.startMenu) {
+                this.startMenu.active = false;
+            }
+            // When active is set to true, it will start listening for mouse events directly, while mouse up events are not yet dispatched
+            // What will happen is that the character will start moving the moment the game starts
+            // Therefore, a delay is needed here
+            setTimeout(() => {
+                if (this.playerCtrl) {
+                    this.playerCtrl.setInputActive(true);
                 }
-                // Directly setting active will directly start monitoring
-                // mouse events, and do a little delay processing
-                setTimeout(() => {
-                    if (this.playerCtrl) {
-                        this.playerCtrl.setInputActive(true);
-                    }
-                }, 0.1);
-                break;
-            case GameState.GS_END:
-                break;
-        }
-        this._curState = value;
+            }, 0.1);
+            break;
+        case GameState.GS_END:
+            break;
     }
-    ```
+}
+```
 
-8. Add event monitoring to the `Play` button. In-order to start the game after clicking the `Play` button, the button needs to respond to click events. Add code that responds to the button click in the `GameManager` script, and click to enter the game's `Playing` state:
+Here we can see in the newly added `init` method, it initializes the runway, activates the start menu, and disable user input at the same time. In any case, we will invoke this method when the game state changes to `GS_INIT`.
 
-    ```ts
-    onStartButtonClicked() {
-        this.curState = GameState.GS_PLAYING;
-    }
-    ```
+And for the first time, accessors like `set curState` allow us to be more secure when setting the game state.
+
+### Add an Event Listener to the Play Button
+
+In order to start the game immediately after clicking the Play button, we need to listen to its click event. Add the following response code in the `GameManager` script as a means to enter the Playing state after the user clicked the button.
```

Comment:
Add the following response code -> Add the following code

response code means the code return from a call.